pkg/scheduler/framework/runtime/framework.go

```diff
@@ -88,6 +89,35 @@ type frameworkImpl struct {
 	parallelizer parallelize.Parallelizer
 }
 
+type instrumentedFilterPlugin struct {
+	framework.FilterPlugin
+
+	cachedMetric compbasemetrics.CounterMetric
+}
+
+func (i *instrumentedFilterPlugin) Filter(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
+	i.cachedMetric.Inc()
+	return i.FilterPlugin.Filter(ctx, state, pod, nodeInfo)
+}
+
+var _ framework.FilterPlugin = &instrumentedFilterPlugin{}
```

Comment:
Nit: I guess we can have a separate file to contain these codes because we may have more wrapped plugins in the future but not strictly, we can do this later if we do.

Reply:
Done.